File: readability.c
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(char text[1000]);
int count_words(char text[1000]);
int count_sentences(char text[1000]);

int main(void)
{
    // Prompt the user for some text
    char text[1000];
    printf("Text: ");
    scanf("%999s", text);

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Compute the Coleman-Liau index
    double L = ((double) letters / words) * 100;
    double S = ((double) sentences / words) * 100;
    double index = 0.0588 * L - 0.296 * S - 15.8;
    float index1 = round(index);
    int index2 = (int)index1;

    // Print the grade level
    if (index2 < 16 && index2 >= 1)
    {
        printf("Grade %i\n", index2);
    }
    else if (index2 >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

int count_letters(char text[1000])
{
    // Return the number of letters in text
    int j = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            j++;
        }
    }
    return j;
}

int count_words(char text[1000])
{
    // Return the number of words in text
    int j = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isspace(text[i]))
        {
            j++;
        }
    }
    return j + 1;
}

int count_sentences(char text[1000])
{
    // Return the number of sentences in text
    int j = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (ispunct(text[i]))
        {
            if (text[i] == '!' || text[i] == '.' || text[i] == '?')
            {
                j++;
            }
        }
    }
    return j;
}

File: readability.py
# Determines grade level of inputted text
def main():
    # Prompt the user for some text
    text = input("Text: ")

    # Count the number of letters, words, and sentences in the text
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Compute the Coleman-Liau index
    L = (letters / words) * 100
    S = (sentences / words) * 100
    index = 0.0588 * L - 0.296 * S - 15.8

    index1 = round(index)
    index2 = int(index1)

    # Print the grade level
    if index2 < 16 and index2 >= 1:
        print(f"Grade {index2}")
    elif index2 >= 16:
        print("Grade 16+")
    else:
        print("Before Grade 1")


def count_letters(text):
    # Return the number of letters in text
    j = 0
    for char in text:
        if char.isalpha():
            j += 1
    return j


def count_words(text):
    # Return the number of words in text
    words_list = text.split()
    return len(words_list)


def count_sentences(text):
    # Return the number of sentences in text
    j = 0
    for char in text:
        if char in ['.', '!', '?']:
            j += 1
    return j


main()
